flex_model/distributed/distributed_state.py

```diff
@@ -195,168 +343,181 @@ def initialize_distributed_state(
         and dp sizes.
     """
     assert (
-        world_size
-        == tensor_parallel_size * pipeline_parallel_size * data_parallel_size
+        dist.is_initialized()
```

Comment:
If a user is running `FlexModel` with one GPU and `FlexModel` distributed requires a process group, then we should initialize and destroy it opaquely to the user.